whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/grs/ArinGrsSource.java

```diff
@@ -84,83 +98,101 @@ public void handleObjects(final File file, final ObjectHandler handler) throws I
-                    }
+            handleLines(reader, new ArinLineHandler(handler));
+        }
+    }
 
-                    final RpslObjectBuilder rpslObjectBuilder = new RpslObjectBuilder(Joiner.on("").join(lines));
-                    for (RpslObject next : expand(rpslObjectBuilder.getAttributes())) {
-                        handler.handle(next);
-                    }
-                }
+    @Override
+    public void handleIrrObjects(final File file, final ObjectHandler handler) throws IOException {
+        try (InputStream is = new GzipCompressorInputStream(new FileInputStream(file))) {
+            final BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.ISO_8859_1));
+            handleLines(reader, new ArinLineHandler(handler));
+        }
+    }
 
-                private List<RpslObject> expand(final List<RpslAttribute> attributes) {
-                    if (attributes.get(0).getKey().equals("ashandle")) {
-                        final String asnumber = findAttributeValue(attributes, "asnumber");
-                        if (asnumber != null) {
-                            final Matcher rangeMatcher = AS_NUMBER_RANGE.matcher(asnumber);
-                            if (rangeMatcher.find()) {
-                                final List<RpslObject> objects = Lists.newArrayList();
-                                int begin;
-                                int end;
-                                try {
-                                    begin = Integer.parseInt(rangeMatcher.group(1));
-                                    end = Integer.parseInt(rangeMatcher.group(2));
-                                } catch (NumberFormatException ex) {
-                                    LOGGER.info(String.format("%s is larger than 32 bit limit in RIPE Database", asnumber));
-                                    return Collections.emptyList();
-                                }
-
-                                for (int index = begin; index <= end; index++) {
-                                    attributes.set(0, new RpslAttribute(AttributeType.AUT_NUM, String.format("AS%d", index)));
-                                    objects.add(new RpslObject(transform(attributes)));
-                                }
-
-                                return objects;
-                            }
-                        }
-                    }
+    private class ArinLineHandler implements LineHandler {
 
-                    return Lists.newArrayList(new RpslObject(transform(attributes)));
-                }
+        private final ObjectHandler objectHandler;
+
+        public ArinLineHandler(final ObjectHandler objectHandler) {
+            this.objectHandler = objectHandler;
+        }
+
+        @Override
+        public void handleLines(final List<String> lines) {
+            if (lines.isEmpty() || IGNORED_OBJECTS.contains(ciString(StringUtils.substringBefore(lines.get(0), ":")))) {
```

Comment:
getFirst instead of get(0)?
It is something new from Java 21

Reply:
I won't refactor existing code, the diff is already big enough and I don't want to introduce any side-effects